Composite/Plugins/Functions/WidgetFunctionProviders/StandardWidgetFunctionProvider/DataReference/PageReferenceSelectorWidgetFunctionBase.cs

```diff
@@ -1,27 +1,36 @@
-using System;
-using System.Xml.Linq;
-using Composite.C1Console.Elements;
 using Composite.Core.Types;
 using Composite.Core.Xml;
+using Composite.Data;
 using Composite.Data.Types;
 using Composite.Functions;
 using Composite.Plugins.Elements.ElementProviders.PageElementProvider;
 using Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.Foundation;
+using System;
+using System.Xml.Linq;
 
 namespace Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.DataReference
 {
     internal abstract class PageReferenceSelectorWidgetFunctionBase : CompositeWidgetFunctionBase
     {
         protected const string CompositeNameBase = CommonNamespace + ".DataReference";
 
-
         private const string HomePageIdParameterName = "HomePageId";
-        protected PageReferenceSelectorWidgetFunctionBase(string compositeName, Type returnType, EntityTokenFactory entityTokenFactory) : base(compositeName, returnType, entityTokenFactory)
+        protected PageReferenceSelectorWidgetFunctionBase(string compositeName, Type returnType, EntityTokenFactory entityTokenFactory)
+            : base(compositeName, returnType, entityTokenFactory)
         {
-            base.AddParameterProfile(
-                new ParameterProfile(HomePageIdParameterName, typeof(Guid?), false,
-                    new ConstantValueProvider(null), new WidgetFunctionProvider(new HomePageSelectorWidgetFunction(entityTokenFactory)), null,
-                    "Filter by Home Page", new HelpDefinition("Use this field to filter by root website. If not set all websites are shown. You can use GetPageId function to get current page")));
+            var boolProvider = StandardWidgetFunctions.GetBoolSelectorWidget(
+                "Enabled: filter pages by active website",
+                "Disabled: no filter, show all websites pages (default)");
+
+            base.AddParameterProfile(new ParameterProfile(
+                HomePageIdParameterName,
+                typeof(bool),
+                false,
+                new ConstantValueProvider(false),
+                boolProvider,
+                "Filter by website",
```

Comment:
Why you renname "Filter by Home Page" to "Filter by website"

Reply:
It seemed for me more clear for understanding. Filter by website => so we can expect to see pages of some certain website. Filter by homepage => so we can expect to see pages which a) have in a **root** a home page, so we will see pages of a certain website or b) have in a **parent**  a home page, so we will see level 1 and level 2 pages. It seemed for me a little bit misunderstanding, but it is just my suggestion. Keep it or turn back?